upgrade/1.2/scripts/upgrade/prepare-assets.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+set -e
+locOfScript=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+. ${locOfScript}/../common/upgrade-state.sh
+. ${locOfScript}/../common/ncn-common.sh $(hostname)
+trap 'err_report' ERR
+# array for paths to unmount after chrooting images
+declare -a UNMOUNTS=()
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --csm-version)
+    CSM_RELEASE="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    --endpoint)
+    ENDPOINT="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    --tarball-file)
+    TARBALL_FILE="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    *)    # unknown option
+    echo "[ERROR] - unknown options"
+    exit 1
+    ;;
+esac
+done
+
+if [[ -z ${CSM_RELEASE} ]]; then
+    echo "CSM RELEASE is not specified"
+    exit 1
+fi
+
+if [[ -z ${TARBALL_FILE} ]]; then
+    # Download tarball from internet
+
+    if [[ -z ${ENDPOINT} ]]; then
+        # default endpoint to internal artifactory
+        ENDPOINT=https://artifactory.algol60.net/artifactory/releases/csm/1.2/
+        echo "Use internal endpoint: ${ENDPOINT}"
+    fi
+
+    # Ensure we have enough disk space
+    reqSpace=80000000 # ~80GB
+    availSpace=$(df "$HOME" | awk 'NR==2 { print $4 }')
+    if (( availSpace < reqSpace )); then
+        echo "Not enough space, required: $reqSpace, available space: $availSpace" >&2
```

Comment:
Minor: it should be a semi-colon or a period after `not enough space`, not a comma.